### PR TITLE
Fix push with local relative

### PR DESCRIPTION
**Pull request by bpeabody**

### CODE REVIEW

**Comment by bpeabody** [comment]
Note that this PR depends on https://github.com/twosigma/git-meta/pull/243

**Comment by bpeabody** [comment]
@novalis ping

File: node/lib/util/git_util.js [hunk 1]
```diff
@@ -138,14 +184,34 @@ exports.isValidRemoteName = co.wrap(function *(repo, name) {
...
+            return null;
+        }
+    });
+    const remote = yield helper();
+
+    if (null !== remote) {
+        return exports.getRemoteUrl(repo, remote);
     }
-    return remote.url();
+    return remote;
```
Comment (novalis) [inline node/lib/util/git_util.js]: return null?

File: node/lib/util/submodule_config_util.js [hunk 2]
```diff
@@ -109,10 +109,22 @@ exports.computeRelativeWorkDir = function (subName) {
...
-    if (0 !== baseUrl.length && baseUrl[baseUrl.length - 1] !== "/") {
+
+    // Make sure ends with trailing "/" so resolution will work with "."
+    // correctly.
+
+    if (!baseUrl.endsWith("/")) {
         baseUrl += "/";
     }
-    return url.resolve(baseUrl, relativeUrl);
+    const res =  url.resolve(baseUrl, relativeUrl);
```
Comment (novalis) [inline node/lib/util/submodule_config_util.js]: nit: is there an extra space after the =?

**Comment by novalis** [comment]
Two minor things -- feel free to fix (or don't change, in the one case, if you prefer) and merge without further approval.

**Comment by bpeabody** [comment]
@novalis addressed "nits" (nice catches, thanks).

**Pull request merged**